src/Hint/NewType.hs

```diff
@@ -138,3 +150,41 @@ dropConsBang x = x
 isUnboxedTuple :: HsType GhcPs -> Bool
 isUnboxedTuple (HsTupleTy _ HsUnboxedTuple _) = True
 isUnboxedTuple _ = False
+
+data WarnNewTypeUn = WarnNewTypeUn
+    { unNewDecl      :: LHsDecl GhcPs
+    , unNewFieldName :: String
+    }
+    | NoWarn
+
+newtypeUn :: LHsDecl GhcPs -> [Idea]
+newtypeUn old
+  | WarnNewTypeUn{unNewDecl, unNewFieldName} <- newTypeWithoutUn old
+  = let
+      note fieldName = [Note $ "Rename field to " <> unNewFieldName]
+      hintName = "Newtype field name prefixed"
```

Comment:
From this name, I wouldn't have guessed what hint this referred to. Can we come upw ith a better name?